packages/babel-parser/src/parser/statement.js

```diff
@@ -1573,9 +1576,12 @@ export default class StatementParser extends ExpressionParser {
     node: N.ClassPrivateProperty,
   ): N.ClassPrivateProperty {
     this.scope.enter(SCOPE_CLASS | SCOPE_SUPER);
+    // [In] production parameter is tracked in parseMaybeAssign
+    this.param.enter(PARAM_);
```

Comment:
I am not sure if I can still understand what I have written at https://github.com/babel/babel/pull/10946/files#diff-a9acee4ca26b68744fe0a1cb5aa03e34R60 three months later.

But this change should be more intuitive as it is copied from the spec:
```
FieldDefinition[Yield, Await]:
  ClassElementName[?Yield, ?Await]  Initializer[In, ~Yield, ~Await]opt
```